plugins/modules/backup_restore_job_info.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: backup_restore_job_info
+version_added: 6.0.0
+short_description: List information about backup restore jobs.
+description:
+    - List detailed information about AWS Backup restore jobs initiated to restore a saved resource.
+author:
+    - Mandar Vijay Kulkarni (@mandar242)
+options:
+  account_id:
+    description:
+      - The account ID to list the restore jobs from.
+    required: false
+    type: str
+  status:
+    description:
+      - Status of restore jobs to filter the result based on job status.
+    required: false
+    choices: ['PENDING', 'RUNNING', 'COMPLETED', 'ABORTED', 'FAILED']
+    type: str
+  created_before:
+    description:
+      - Specified date to filter result based on the restore job creation datetime.
+      - If specified, only the restore jobs created before the specified datetime will be returned.
+    required: false
+    type: str
+  created_after:
+    description:
+      - Specified date to filter result based on the restore job creation datetime.
+      - If specified, only the restore jobs created after the specified datetime will be returned.
+    required: false
+    type: str
+  completed_before:
+    description:
+      - Specified date to filter result based on the restore job completion datetime.
+      - If specified, only the restore jobs created before the specified datetime will be returned.
+    required: false
+    type: str
+  completed_after:
+    description:
+      - Specified date to filter result based on the restore job completion datetime.
+      - If specified, only the restore jobs created after the specified datetime will be returned.
+    required: false
+    type: str
+extends_documentation_fragment:
+  - amazon.aws.common.modules
+  - amazon.aws.region.modules
+  - amazon.aws.boto3
+"""
+
+EXAMPLES = r"""
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: List all restore jobs
+  amazon.aws.backup_restore_job_info:
+
+- name: List restore jobs based on Account ID
+  amazon.aws.backup_restore_job_info:
+    account_id: xx1234567890
+
+- name: List restore jobs based on status and created_before time
+  amazon.aws.backup_restore_job_info:
+    status: completed
+    created_before: "2023-02-25T00:05:36.309Z"
+"""
```

Comment:
I guess this module should also support to gather information on a specific restore job also (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/backup/client/describe_restore_job.html) and not only listing jobs.

@tremble what do you think?

Reply:
+1 describing is generally useful functionality